### ISSUE TITLE
Update docs regarding parameter remapping

### ISSUE THREAD

**Issue by christophfroehlich** [ros-controls/ros2_control#1765]
>    >  Thanks for the detailed summary of your issue. @saikishor Isn't the part in [docs](https://control.ros.org/humble/doc/ros2_control/controller_manager/doc/userdoc.html#using-the-controller-manager-in-a-process) outdated after #1694?
> 
> Yes, it is outdated as now such individual node remapping is no longer needed, even though the same trick will still work. Should we cleanup this part of the docs?

_Originally posted by @saikishor in https://github.com/ros-controls/ros2_control/issues/1704#issuecomment-2304115262_
